manifests/state-nic-feature-discovery/030-nic-feature-discovery-ds.yaml

```diff
@@ -0,0 +1,80 @@
+# 2023 NVIDIA CORPORATION & AFFILIATES
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: nic-feature-discovery-ds
+  namespace: {{ .RuntimeSpec.Namespace }}
+  labels:
+    tier: node
+    app: nic-feature-discovery
+    name: nic-feature-discovery
+spec:
+  selector:
+    matchLabels:
+      name: nic-feature-discovery
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        tier: node
+        app: nic-feature-discovery
+        name: nic-feature-discovery
+    spec:
+      terminationGracePeriodSeconds: 10
+      {{ if eq .RuntimeSpec.OSName "rhcos" }}
+      serviceAccountName: nic-feature-discovery
+      {{- end }}
+      tolerations:
+      - key: nvidia.com/gpu
```

Comment:
It looks like we are missing support for custom tollerations here
```
        {{- if .Tolerations }}
        {{- .Tolerations | yaml | nindent 8 }}
        {{- end }}
```

Reply:
fixed !